docs: touchscreen-settings-available-features

jwwojak opened the pull request:
# Overview

Updates the Flex manual chapter **Touchscreen** > **Robot Settings**. This lists and describes the currently available features on that part of the touchscreen.

Sandbox: https://sandbox.docs.opentrons.com/docs-touchscreen-settings-revisions/flex/touchscreen/settings/

RTC-896

Maybe for January.

## Test Plan and Hands on Testing

Check images and links.

## Changelog

This PR introduces some changes to the existing doc to take advantage of MkDocs features (e.g. tables of contents) that weren't available when this was first created.

- Revised intro paragraph

- Changed image to fade out and show _fewer_ options from the settings screen. Why? Gets us out of the business of maintaining an image when features change. Just update text and show only enough of an image so readers recognize/know where they are on the touchscreen. Text changes are easier than image updates.

- Keeps the topical section organization but now uses H2 and H3 headers to show all the settings on the page-level TOC.
    - H2 headers are categories
    - H3 headers are individual settings

In the current version, the TOC is somewhat opaque with category headers only. With H3s we can now list and show all the settings, which also eliminates the need for a full image at the beginning.

## Review requests

- ~~Still unsure about one particular setting: Disabling some Stacker x y axis detection. Checking on it. A WTF item.~~
- Does the new organization/structure work. I think it does but is there an even better way to move things around?

## Risk assessment

- Low, docs only.

jwwojak committed:
Not a great image

Testing ideas

jwwojak committed:
Coffee interval commit

jwwojak committed:
Lunch save

jwwojak committed:
Afternoon commit and PR create

jwwojak committed:
Might be ready for PR

jwwojak committed:
Final adjustments and clarifications

emilyburghardt commented on docs/flex/docs/images/touchscreen-settings-fade.png:
a thought about the fade in the image: 

I agree it will save us some time to no longer update the touchscreen image every time there's a change, especially as the list gets longer. I do think the fade effect could be a little confusing since it doesn't actually look like this on the touchscreen, right? I found myself asking this and going to look for myself 😆 

I suggest adding a sentence to explain this. I like the caption that states _examples_ of configurable items. Maybe also a sentence before or after like `Scroll down to view all available settings for your Flex.`

jwwojak replied:
Sure. I can get rid of that and add a caption.

jwwojak replied:
Removing image. Will get a complete image. Checking w/ Felix & Sue for latest.

emilyburghardt commented on docs/flex/docs/touchscreen/settings.md:
```diff
@@ -2,60 +2,107 @@
+
+### Disable Stacker sensors for labware detection in z- and x-axis
+
+Controls the _Time of Flight_ (ToF) sensor in the Flex Stacker. By default, the ToF sensor detects if labware is loaded in the Stacker before this external module attempts to dispense or store it.
```
do we have any more guidance on what `specialized labware` used in the Stacker is? 

is this custom labware definitions? or can some of our LL labware trigger these detection errors?

jwwojak replied:
Maybe ~~specialized~~. The feature is for any labware that works in the stacker but is causing detection problems. Could be supported or unsupported, as long as it works.

emilyburghardt reviewed (commented):
general comments to open a discussion!

ecormany commented on docs/flex/docs/touchscreen/settings.md:
```diff
@@ -2,60 +2,107 @@
+
+### Disable Stacker sensors for labware detection in z- and x-axis
+
+Controls the _Time of Flight_ (ToF) sensor in the Flex Stacker. By default, the ToF sensor detects if labware is loaded in the Stacker before this external module attempts to dispense or store it.
```
_time of flight_ can and should be lowercase even if the abbreviation _ToF_ isn't.

ecormany commented on docs/flex/docs/touchscreen/settings.md:
```diff
@@ -2,60 +2,107 @@
 
 !!! note
-    There are separate privacy controls in the Opentrons App. Turning sharing on or off from the touchscreen only affects data collected and sent by the robot. Your laptop or desktop computer will still automatically share data if this feature is enabled in the Opentrons App.
+    When disabled, the Stacker will always try to store or dispense labware, even if it is empty.
```
Something seems off here…I think the text is trying to combine two concepts:
- The Stacker will dispense labware even if it is empty (and therefore shouldn't)
- The Stacker will store labware even if it is full (and therefore shouldn't)

ecormany commented on docs/flex/docs/touchscreen/settings.md:
```diff
@@ -2,60 +2,107 @@
+
+## Camera
+
+Every Flex comes equipped with a [built-in camera][camera-features-and-controls], which is off by default. Camera options include on/off settings for still photographs, video, and on-error image capture.
```
We walk through some other settings in detail, and there's a lot on the Camera Preferences screen (once you flip the overall setting to Enabled). Should we say more here?

jwwojak committed:
Time of flight description revision

jwwojak committed:
Remove H2 sections, put in alpha order

Needs new image

jwwojak committed:
Merge branch 'edge' into docs-touchscreen-settings-revisions

jwwojak committed:
Remove bad image, re-use original, alpha order

Put settings in alphabetical order.

jwwojak committed:
ToF definition adjustment

jwwojak committed:
Add header, use text to future-proof image

So we don't need to update the image. Explain "yours can look different" and keep the definitions up to date. This is good enough.

jwwojak committed:
Got good Settings image!!!!

Reduced size 80%

emilyburghardt commented on docs/flex/docs/touchscreen/settings.md:
```diff
@@ -2,36 +2,56 @@
+
+## Settings example
+
+The image below shows an example of the Settings section on the Flex touchscreen. Because Opentrons frequently updates the robot's software, the options on your screen may be different. Scroll through on the touchscreen to see all the currently available settings. Refer to the [Settings list section](#settings-list) for definitions of the currently released features.
```
suuuuuper nitpick: 
```suggestion
Refer to the [Settings list section](#settings-list) below for definitions of the currently released features.
```

Only because I read this, thought to myself "oh where is that how handy" and clicked on it, then was surprised I was redirected down on the same page.

jwwojak replied:
Yes, that is an excellent nit to pick! Thanks!

emilyburghardt commented on docs/flex/docs/touchscreen/settings.md:
```diff
@@ -2,36 +2,56 @@
-- **Robot System Version:** See the current version of the robot software or check for updates. If Flex has already automatically checked for updates and found one, this item will have an "Update available" badge in the settings list.
+Batch delete certain types of information from the robot, such as calibrations, run history, or protocols.
+
+### Disable Stacker sensors for labware detection in z- and x-axis
```
again, an incredible nitpick. 

Is including "z- and x-axis" in the section title confusing/too much for users? I guess I'm thinking (too much) about the labware in the Stacker really just being in the Z axis (especially if you're looking down from the top of the stack or up from the bottom). 

totally fine to leave either way, but since we don't mention z- and x-axis in the Stacker later in this short section, might not be needed.

jwwojak replied:
We have to include that because this is the verbatim text on the button label. Gotta use it, unfortunately. Agree that it is confusing or awkward.

emilyburghardt reviewed (approved):
Left a couple super nitpick comments for consideration. but LGTM!

jwwojak committed:
Apply suggestions from code review

Committing suggestions from reviewers.

Co-authored-by: emilyburghardt <emily.burghardt@opentrons.com>
